aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,295 @@
+```csharp
+var options = new RewriteOptions()
+    .AddRedirectToHttps(301, 5001);
+
+app.UseRewriter(options);
+```
+Use `AddRedirectToHttpsPermanent()` to redirect insecure requests to the same host and path with secure HTTPS protocol (**https://** on port 443). The middleware will set the status code to 301 (Moved Permanently).
+
+The sample application is capable of demonstrating how to use `AddRedirectToHttps()` or `AddRedirectToHttpsPermanent()`. Add the extension method to the `RewriteOptions()`. Make an insecure request to the application at any URL. In order to see the response in a browser, you will probably need to dismiss a browser security warning that the self-signed certificate is untrusted.
+
+Original Request using `AddRedirectToHttps(301, 5001)`: **http://localhost:5000/secure**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_redirect_to_https.png)
+
+Original Request using `AddRedirectToHttpsPermanent()`: **http://localhost:5000/secure**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_redirect_to_https_permanent.png)
+
+#### URL Rewrite
+Use `AddRewrite()` to create a rules for rewriting URLs. The first parameter will contain your regex for matching on the incoming URL path. The second parameter is the replacement string. The third parameter, `skipRemainingRules: {true|false}`, will indicate to the middleware whether or not to skip additional rewrite rules if the current rule is applied.
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=3)]
+
+Original Request: **http://localhost:5000/rewrite-rule/1234/5678**
+
+![Browser window with Developer Tools tracking the request and response](url-rewriting/_static/add_rewrite.png)
+
+The first thing you will notice in the regex is the carat (`^`) at the beginning of the expression. This means that matching should be attempted starting at the beginning of the URL path.
+
+In the earlier example with the redirect rule, `redirect-rule/(.*)`, there is no carat at the start of the regex; therefore, any characters may precede **redirect-rule/** in the path for a successful match.
+
+URL | Match
+--- | :---:
+**http://localhost:5000/redirect-rule/1234/5678** | Yes
+**http://localhost:5000/my-cool-redirect-rule/1234/5678** | Yes
+**http://localhost:5000/foobarredirect-rule/1234/5678** | Yes
+
+The rewrite rule, `^rewrite-rule/(\d+)/(\d+)`, will only match paths if they start with **rewrite-rule/**. Notice the difference in matching between the rewrite rule below and the redirect rule above.
```

Comment:
It took me a long time to realize there's a "match" column because it shows up so far off to the right away from the URL. But the table is coded right so I don't have an easy fix for it.  Could make column 1 right-justified but that might look odd.